### ISSUE TITLE
Selecting photo for new tag sometimes replaces selected photo for found tag

### ISSUE THREAD

**Issue by ykyu** [KenEucker/biketag-website#154]
I've run into this a few times now playing Seattle's game and I think others have too, but it's pretty inconsistent so I'm not sure how to troubleshoot further.

Sometimes when selecting a photo for the new tag (B), it will both show up in the preview box for the new tag _and_ the preview box for the old tag/proof photo (A). Re-selecting the photo for A is then necessary or else the photo for B will show up in both spots upon upload. Sometimes after re-selecting the photo for A, I _also_ have to re-select the photo for B, which will either go blank or show the photo for A.

Basically it ends up have I to re-verify that both photos are correct several times before uploading. This has happened maybe 2-3 times in my last 10 tags? Hard to say, but I'll try to keep better track of it moving forward. I definitely had the issue when trying to upload #97 yesterday.

Using Chrome (v86) on Android 10.

**Comment by ykyu** [comment]
Actually, I just tested this a few times from desktop. It's still not a 100% reproduction rate, but I can hit the bug fairly consistently if I:

1. Upload a photo for A (proof tag).
2. Upload a photo for B (new tag).
3. Delete the photo for B.
4. Select a new photo for B.

The newly select for B then replaces the photo for A, but no new photo for B appears to have been selected. If you select a photo for B again, then it will show up.

If you repeat the process -- deleting the photo for B and re-selecting it -- it seems to replace the photo for A every other time.

So uh. That's weird.

**Comment by KenEucker** [comment]
@ykyu Thank you so much for creating this issue and for your testing! This is something that we've been trying to track down and solve for a while but haven't had many clues as to why it's occurring. I will look into this and report back on testing, and hopefully we can get this resolved soon.

**Comment by KenEucker** [comment]
@ykyu I wasn't able to reproduce this issue in Chrome, but I did push a change that may resolve the issue. Can you test it and see if you're able to reproduce it like you were able to previously?

**Comment by ykyu** [comment]
Just to be clear, I'm testing via the Seattle biketag live site -- I don't have a local copy running at the moment.

Result 1: Select photo for A; it shows up. Select photo (B1) for B; it shows up where A should be. Select another photo (B2) for B; the originally selected B1 photo still shows up for A. Select a third photo for B. No change.

Result 2: Select photo for A; it shows up. Select photo for B; it does not show up in either the A spot or the B spot. Select photo for B. No change. Select photo for B; it shows up where it should.

Result 3: Select photo for A; it shows up. Select photo (B1) for B; it shows up where A should be. Select another photo (B2) for B; it shows up where B should be. Select photo for A; the previous image (B1) disappears, but the newly select A does not replace it. Select photo for A; no change. Select photo for A; no change.

Result 4: Select photo for A; it shows up. Select photo (B1) for B; it shows up where A should be. Select another photo (B2) for B; it shows up where B should be. Select photo for A; it shows up where A should be.

Result 5: Select photo for A; it shows up. Select photo for B; it shows up. Close out select photo for B. Select new photo for B; it shows up where A goes. (This was the original reproduction case.) Select another photo for B; no change. Select another photo for B; it shows up where A goes.

I tested on both desktop Chrome and Firefox. Chrome had way more variants; Firefox most consistently led to result 5, but the error is definitely still there.

**Comment by KenEucker** [comment]
@ykyu I tried to reproduce these cases but was unable to. Both using Chrome and Firefox on OS X, as well as Chrome on Android 11.

Here is a Loom of the behavior I see every time I use the form:
https://www.loom.com/share/f021da2fb89141859df362fae3847656

**Comment by ykyu** [comment]
@KenEucker Sorry it took so long for me to get back to this!

https://www.loom.com/share/9b8d6e7fb9e44913bb9b8842f5ff81a6

For the first minute, it seems that selecting photo B works, but if I x out the photo and try to select another one, it won't launch the file selection. (I should've shared the whole desktop to make this more obvious, oof.) Around 1:45 you can finally see me click the selection for photo B, but it updates photo A instead.

**Comment by KenEucker** [comment]
In my testing this last month I've noticed that there are issues with selecting images and clicking out of the text boxes simultaneously going into another image upload. There are two items in the work that I believe will solve this issue:

A new form for creating posts that separates the fields by "cards", so you wouldn't go from entering an image into a textbox immediately anymore.
And
A "queue" where users can upload images one at a time, the moment they first arrive, and snap the mystery tag on their phone.

I don't think that I will specifically address this bug before completing the items above, so I will circle back then and see if this issue is resolved. Thanks again, @ykyu !

**Comment by ykyu** [comment]
Follow-up: Since the updates, I have only run into this issue once and that was the other day when Imgur was (apparently) down and I was trying to submit [proof for 182 and the new tag 183](https://www.reddit.com/r/CycleSeattle/comments/ks0rv6/biktag_183/). I think I tried to submit 3-4 times and this issue happened every time after my first attempt. Weird and maybe not reproducible, but I figured I'd mention it!

**Comment by KenEucker** [comment]
@ykyu thanks for mentioning and for the follow-up! I do want to circle back and try to resolve this at some point but I do have a feeling that we are getting closer to it being a non-issue. Of course, if anything changes do come back and let me know.

Do It For The Ride,

**Comment by KenEucker** [comment]
This issue still persists. :(

**Comment by wong-a** [comment]
I see this too. It happens almost all the time on Samsung Internet on Android.

**Comment by KenEucker** [comment]
As this project is now deprecated, I am closing out all outstanding issues. Please see the new BikeTag App project here: https://keneucker/biketag-vue.

The biketag-website project is now sunsetting.
